app/plugin_api_test.go

```diff
@@ -93,6 +95,11 @@ func setupMultiPluginApiTest(t *testing.T, pluginCodes []string, pluginManifests
 	}
 
 	app.SetPluginsEnvironment(env)
+	app.UpdateConfig(func(cfg *model.Config) {
+		for _, pluginId := range pluginIds {
+			cfg.PluginSettings.PluginStates[pluginId] = &model.PluginState{Enable: true}
+		}
+	})
```

Comment:
`.PluginSettings.PluginStates` seams to like an "internal" representation that tests should not care about. I'm 1/5 that the "public" API `app.EnablePlugin(pluginId)` should be used instead even it it mostly does the same as setting `cfg.PluginSettings.PluginStates[pluginId]`.

Reply:
Agreed, though in this method we're also injecting our own instance of the plugin environment vs. using `InstallPlugin` normally. My gut is this needs to be fixed holistically.

Reply:
I missed own environment point. Makes sense to leave it as it is.